examples/13-modify-response.php

```diff
@@ -0,0 +1,26 @@
+<?php
+use React\Http\Server;
+use React\Http\Response;
+use React\EventLoop\Factory;
+use Psr\Http\Message\ServerRequestInterface;
+
+require __DIR__ . '/../vendor/autoload.php';
+
+$loop = Factory::create();
+
+$server = new Server([
+    function (ServerRequestInterface $request, callable $next) {
+        return $next($request)
+            ->then(function(\Psr\Http\Message\ResponseInterface $response) {
```

Comment:
Good catch! I'm all for addressing this and also improving the documentation here! :+1: 

Practically this should work just fine. However, AFAICT this is currently an implied assumption and not something that is guaranteed by either our interfaces or the documentation. If you were to wrap this in a `Promise\resolve()` call, then this would make perfect sense. Perhaps, this should even be recommended as it eases how consumers see middleware request handlers?

There's some obvious potential for performance improvements here, as the promise wrapping is entirely unneeded. Should we guarantee this is always the case or rather add documentation for possible return values?

Reply:
I can suggest, that you should explicitly cover this moment about `MiddlewareRunner` in the documentation https://github.com/reactphp/http/blob/master/src/Io/MiddlewareRunner.php#L47
otherwise we simply wrap a promise into another promise.

Also, I've removed short array syntax, as `composer.json` requires support for PHP 5.3